test/jit/test_tracer.py

```diff
@@ -2279,3 +2279,27 @@ def forward(self, input: Dict[str, Tuple[torch.Tensor, torch.Tensor, torch.Tenso
         traced_module = torch.jit.trace(eager_module, input1)
         self.assertEqual(traced_module(input1), eager_module(input1))
         self.assertEqual(traced_module(input2), eager_module(input2))
+
+    def test_trace_returning_complex_dict(self):
+        """Tracing over a module returning a complex dictionary should work.
+        The dictionary can should be able to contain other containers (like a tuple) recursively.
+        """
+        class ReturnsDict(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(
+                self, t: torch.Tensor
```

Comment:
have two inputs, one for key, one for value? sometimes, the constant just hides the problem.